nft_ingester/src/program_transformers/token_metadata/v1_asset.rs

```diff
@@ -296,25 +296,33 @@ pub async fn save_v1_asset<T: ConnectionTrait + TransactionTrait>(
             txn.execute(query).await?;
         }
     }
-    txn.commit().await?;
-    let creators = data.creators.unwrap_or_default();
-    if !creators.is_empty() {
-        let mut creators_set = HashSet::new();
-        let existing_creators: Vec<asset_creators::Model> = asset_creators::Entity::find()
+
+    // checks if the creator to index is outdated. This assumes that all creator rows with same AssetId have the same SlotUpdated
+    let creator_outdated = asset::Entity::find()
+        .filter(
+            Condition::all()
+                .add(asset::Column::Id.eq(id.to_vec()))
+                .add(asset::Column::SlotUpdated.gte(slot_i)),
+        )
+        .one(conn)
+        .await?
+        .is_some();
+    if !creator_outdated {
+        // delete old creators, delete must come first to avoid unique assetId/creator and assetId/position constraint violations
```

Comment:
Be more clear as to why we are deleting creators. e.g, because we creators can be removed and the cleanest way to handle edge cases is to "reset" the entire list of creators.